src/components/SettingsModal/index.jsx

```diff
@@ -51,15 +52,24 @@ export default class SettingsModal extends React.Component {
   }
 
   handleSubmit(event) {
-    /** Handle a click on the "Save" button, which updates the parent's state */
+    /** Handle a click on the "Save" button.
+      *
+      * Updates the parent's state and persistent store.
+    */
     event.preventDefault();
     this.props.saveSettings(this.state.localSettings);
     this.setState({show: false});
   }
 
+  /** Handle a click on the "Reset Defaults" button, which updates the parent's state */
+  handleDefault(event) {
+    event.preventDefault();
+    this.props.setDefaultSettings();
```

Comment:
It's a minor thing but I think the form buttons here are a bit non-orthogonal because:
1. "reset" gets the `app` state out of sync with the db by calling the parent's `setState` but not calling the store's "save". and
2. "reset" kinda acts like a "submit" button because it updates parent state and applies the changes immediately. So a subsequent "cancel" click doesn't actually throw away the changes.

Do you think the reset button should apply the settings immediately (by updating the `app` state), or only change the display values and then require a "save" click to actually apply the settings? I think the latter might be a little more conventional.

Reply:
> or only change the display values and then require a "save" click to actually apply the settings?

Yea this sounds right.